packages/binding-file/src/file-client.ts

```diff
@@ -17,51 +17,39 @@
  * File protocol binding
  */
 import { Form, SecurityScheme } from "@node-wot/td-tools";
-import { ProtocolClient, Content, createLoggers } from "@node-wot/core";
+import { ProtocolClient, Content, createLoggers, ContentSerdes } from "@node-wot/core";
 import { Subscription } from "rxjs/Subscription";
 import fs = require("fs");
-import path = require("path");
+import { fileURLToPath } from "node:url";
 
-const { debug, warn } = createLoggers("binding-file", "file-client");
+const { debug } = createLoggers("binding-file", "file-client");
 
 export default class FileClient implements ProtocolClient {
     public toString(): string {
         return "[FileClient]";
     }
 
-    public async readResource(form: Form): Promise<Content> {
-        const filepath = form.href.split("//");
-        const resource = fs.createReadStream(filepath[1]);
-        const extension = path.extname(filepath[1]);
-        debug(`FileClient found '${extension}' extension`);
-        let contentType;
-        if (form.contentType != null) {
-            contentType = form.contentType;
-        } else {
-            // *guess* contentType based on file extension
-            contentType = "application/octet-stream";
-            switch (extension) {
-                case ".txt":
-                case ".log":
-                case ".ini":
-                case ".cfg":
-                    contentType = "text/plain";
-                    break;
-                case ".json":
-                    contentType = "application/json";
-                    break;
-                case ".jsonld":
-                    contentType = "application/ld+json";
-                    break;
-                default:
-                    warn(`FileClient cannot determine media type of '${form.href}'`);
-            }
-        }
+    private async readFromFile(filePath: string, contentType: string) {
+        debug(`Reading file of Content-Type ${contentType} from path ${filePath}.`);
+        const resource = fs.createReadStream(filePath);
         return new Content(contentType, resource);
     }
 
+    public async readResource(form: Form): Promise<Content> {
+        const filePath = fileURLToPath(form.href);
+        const contentType = form.contentType ?? ContentSerdes.DEFAULT;
```

Comment:
I know this is correct behavior, but it might be error-prone... people will try to serve files without expressing the content-type thinking that the client could figure it himself. Anyhow, just a comment no action is required here.

Reply:
Maybe we can add a debug comment "saying that"... a la
... non contentType set, picked JSON by default

Reply:
Tried to incorporate this in 14c178039c2537b12c97be60c80cb53dfa14270b, let me know if that works for you :)

In general, I was wondering if we should maybe move the default Content-Type determination to the `Form` class (maybe as part of a `get` method) to have a consistent behavior across the codebase. I would make an initial proposal in a follow-up PR that we could discuss there if that's okay for you.